product_visible_discount_extension/models/sale.py

```diff
@@ -0,0 +1,206 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2015 Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.exceptions import ValidationError
+from openerp import (
+    models,
+    fields,
+    api,
+)
+
+
+def dummy_search(self, *a, **ka):
+    return []
+
+
+class SaleOrderLine(models.Model):
+
+    """
+    Compute the value of visible_discount and price_unit.
+
+    As those two fields are mutual, we have to store a value in a
+    temp variable.
+
+    Any change to visible_discount will trigger onchanges events in that order:
+        _onchange_visible_discount
+        _onchange_price
+
+    Any change to price_unit will trigger onchange events in that order:
+        _onchange_price
+        _onchange_visible_discount
+    """
+
+    _name = 'sale.order.line'
```

Comment:
When you use _inherit is not necessary to add _name unless you want to create a different model.